### PR TITLE
Backport Issue 3984 (.NET6 support++) to v3.13 branch

### LINKED ISSUE #4046: Backport .NET6 test infra + build improvements from #3984 onto v3.13 branch
As noted by @manfred-brands (https://github.com/nunit/nunit/issues/4036#issuecomment-1054919618)
We added .NET6 to our testing matrix on main in #3984 along with some nice build improvements to reduce duplication of specified runtime/library build targets. I think this could be a good candidate to backport to v3.13 as well as it will help keep the testing and tooling between the two consistent

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #4046

Upgrade to .NET 6.0.100 SDK
Use dotnet core build also on Windows
Unify frameworks

### CODE REVIEW

**Comment by manfred-brands** [comment]
MacOS test fails with a timeout, the test `LargeStringCollectionsInSameOrder` exceeds 500ms. It runs in 39ms on my machine.

The existing timeouts are defined like:
```csharp
        // The following tests are each running in 14ms to 46ms on my machine. Based on that,
        // warn at 100ms and fail at 500ms
        const int LARGE_COLLECTION_WARN_TIME = 100;
        const int LARGE_COLLECTION_FAIL_TIME = 500;
```

Changing this seems outside scope.

**Comment by stevenaw** [comment]
@manfred-brands I recall that being a bit of a finnicky test which anecdotally appears to fail sometimes if the cloud VMs are under high utilization. It looks like they passed just fine on GH actions

File: src/NUnitFramework/Directory.Build.targets
Comment (stevenaw) [inline src/NUnitFramework/Directory.Build.targets:6]: Question: This is a small difference between this here and it in Directory.Build.props in https://github.com/nunit/nunit/pull/3988/files, but I think I recall you saying in a different PR or issue that this is to fix an issue related to when `TargetFrameworkIdentifier` is initially defined, is that right?

File: src/NUnitFramework/nunitlite-runner/nunitlite-runner.csproj
Comment (stevenaw) [inline src/NUnitFramework/nunitlite-runner/nunitlite-runner.csproj:40]: Similar question here, I think this helps fix something?

Comment (manfred-brands) [reply src/NUnitFramework/nunitlite-runner/nunitlite-runner.csproj:40]: In SDKs pre .NET 5 one had to select different Sdk projects if one wanted to use Forms or WPF.
This was [unified in NET 5](https://docs.microsoft.com/en-us/dotnet/core/compatibility/sdk/5.0/sdk-and-target-framework-change) so that this is no longer needed.
But as our library doesn't use either, I'm not sure why it was ever in there.
This was removed in master (nunit4) a year ago by Rob. Commit hash 3727fce5f031acb0179c591f455b65703747a9d4.

File: src/NUnitFramework/testdata/nunit.testdata.csproj
Comment (stevenaw) [inline src/NUnitFramework/testdata/nunit.testdata.csproj:13]: I'm not familiar with this change, but I'm assuming it fixes something :)

**Review by stevenaw — APPROVED** [review]
Thanks @manfred-brands this looks good to me!
Most of my review focused on comparing branches and PRs. I'd appreciate a second pair of eyes on this

**Comment by mikkelbu** [comment]
I'll be the second pair of eyes :). I'll take a look at it this weekend (hopefully today).

File: build.cake
Comment (mikkelbu) [inline build.cake:215]: We no longer have any entries in `NetCoreTests` that ends with `windows`.

File: nuget/nunitlite/nunitlite.nuspec
Comment (mikkelbu) [inline nuget/nunitlite/nunitlite.nuspec:54]:
```suggestion
      </group>
```

File: nuget/nunitlite/nunitlite.nuspec
Comment (mikkelbu) [inline nuget/nunitlite/nunitlite.nuspec:81]:
```suggestion
    <file src="bin/net6.0/nunitlite.pdb" target="lib\net6.0" />
```

**Review by mikkelbu — APPROVED** [review]
Also looks good to me and a nice simplification in multiple files.

I agree with @stevenaw comments and have some additional whitespace nit-pick.

**Comment by mikkelbu** [comment]
I tried to see if I could retrigger the build in azure devops, but the build is no longer there, and the branch does not exists in the nunit project, and the commit didn't work, so I'm not sure I can retrigger it.

File: src/NUnitFramework/testdata/nunit.testdata.csproj
Comment (manfred-brands) [reply src/NUnitFramework/testdata/nunit.testdata.csproj:13]: It fixes consistency. nunit.framework.csproj was compiled with net35;net40;net45.
Other assemblies had various other version. They are now all aligned. The condition includes some Microsoft.Bcl packages for functionality that was added to the framework in 4.6

File: build.cake
Comment (manfred-brands) [reply build.cake:215]: This is the same as in main branch. It protects against someone updating `NUnitRuntimeFrameworks` in `Directory.Build.props`. We could remove the whole condition in both branches.

File: src/NUnitFramework/Directory.Build.targets
Comment (manfred-brands) [reply src/NUnitFramework/Directory.Build.targets:6]: Done.

**Comment by manfred-brands** [comment]
I think I have addressed all issues.

**Comment by manfred-brands** [comment]
Rebased, is there anything holding this up being merged?

**Comment by stevenaw** [comment]
Thanks @manfred-brands ! Looks like it built. LGTM 🎉

**Pull request merged**
